guidelines/monai-application-package.md

```diff
@@ -1,498 +1,650 @@
+    - The default value `0` SHALL be assumed when not provided.
 
-    - When the value is a relative file-system path (first character is not `/`), it is relative to the Application's working directory.
+  - The Application Manifest SHALL define how often to perform the readiness probe (`/etc/holoscan/app.json#readiness.periodSeconds`).
 
-    - When the value is an absolute file-system path (first character is `/`), the file-system path is used as-is.
+    - When not provided, the default value `10` SHALL be assumed.
 
-  - When not provided the default value `input/` SHALL be assumed.
+  - The Application Manifest SHALL define the number of seconds after which the probe times out (`/etc/holoscan/app.json#readiness.timeoutSeconds`)
 
-- Application Manifest SHOULD define input data formats supported by the Application (`/etc/monai/app.json#input.formats`).
+    - When not provided, the default value `1` SHALL be assumed.
 
-- Application Manifest SHOULD define output path, relative to the working directory used by the Application (`/etc/monai/app.json#output.path`).
+  - The Application Manifest SHALL define the number of times to perform the probe before considering the service is not ready (`/etc/holoscan/app.json#readiness.failureThreshold`)
 
-  - The input path SHOULD be a relative file-system path to a directory or folder.
+    - The default value `3` SHALL be assumed when not provided.
 
-    - When the value is a relative file-system path (first character is not `/`), it is relative to the Application's working directory.
+- The Application Manifest SHOULD configure a check to determine whether or not the application is "live" or not.
 
-    - When the value is an absolute file-system path (first character is `/`), the file-system path is used as-is.
+  - The Application Manifest SHALL define the type of probe to be performed (`/etc/holoscan/app.json#liveness.type`).
 
-  - When not provided the default value `output/` SHALL be assumed.
+    - Possible values include `tcp`, `grpc`, `http-get`, and `command`.
 
-- Application Manifest SHOULD define output data format produces by the Application (`/etc/monai/app.json#output.format`).
+  - The Application Manifest SHALL define the probe commands to execute when the type is `command` (`/etc/holoscan/app.json#liveness.command`).
 
-- Application Manifest SHOULD define any timeout applied to the Application (`/etc/monai/app.json#timeout`).
+    - The data structure is expected to be an array of strings.
 
-  - When not provided the default value `0` SHALL be assumed.
+  - The Application Manifest SHALL define the port to perform the liveness probe when the type is `grpc`, `tcp`, or `http-get`. (`/etc/holoscan/app.json#liveness.port`)
 
-  - This value can be overridden by the top level executor, such as MONAI Deploy Application Server.
+    - The value provided must be a valid port number ranging from 1 through 65535.
```

Comment:
Same about ports 1024 to 65535.

Reply:
Added a note `Please note that port numbers below 1024 are root-only privileged ports.`